### DISCUSSION TITLE
Is gdsfactory capable of creating waveguides in type of Path rather than Polygon?

### DISCUSSION THREAD

**Discussion by iniytwte** [gdsfactory/gdsfactory#2171] [Q&A] [answered]
In Klayout I can draw a waveguide using Path, but I can't find any way to create Path with gdsfactory. It seems that all geometries are built with Polygon. 

In gdstk Path can be created using FlexPath or RobustPath. I wonder is there any method to create Path with gdsfactory or kfactory?

**Comment by joamatab** [comment, marked as answer]
yes, you are correct,

what would be the advantage of path over polygon?

we were thinking about exposing the Path functionality from gdstk but haven't found a need for it

**Reply by tvt173** [reply]
also, polygons give more flexibility and certainty than paths, i.e. for tapered and curvilinear sections. you can see that even the RobustPath of gdstk actually renders as polygons in some situations rather than paths for this reason (https://heitzmann.github.io/gdstk/gettingstarted.html#paths)

that is to say, GDSII paths will not work to replace polygons in all situations, so the approach of gdsfactory has been to just always use polygons for consistency. as Joaquin says, we haven't found a compelling reason to use Paths instead, but let us know if you have one

**Reply by iniytwte** [reply]
thanks for replying. 

1) i'm not sure about this: the file size of a gds layout of Polygons may be larger than that of Path, as waveguide defined by Polygons contains more points (precisely, double) than Path? especially for dense, long waveguides like spiral delay line

2) width of Paths can be modified directly in Klayout, which might be quite convenient for minor modification of the final layout sometime

except for these considerations, i highly agree that Polygons can avoid many problems in Paths

**Reply by Chuplin** [reply]
I have the same question. Currently, I am using the fixed beam moving stage (FBMS) mode in EBL, which requires patterns to be in PATH rather than POLYGON. If it is not possible to create waveguides in the Path, is there a way to convert a polygon structure to the path format after it has been created?

**Reply by jonbenfri** [reply]
I just came across this issue with an FBMS EBL system requiring Path elements in GDSII.

The solution here worked for me:
https://github.com/gdsfactory/gdsfactory/discussions/2865#discussioncomment-9811069

```python
path = gf.kdb.DPath(pts=[gf.kdb.DPoint(x,y) for x,y in my_list_of_tuple_points], width=path_wodth_in_um)
c.shapes(LAYER.WG).insert(path)
```
or
```python
path = gf.kdb.DPath(pts=my_list_of_tuple_points, width=path_width_in_um)
c.shapes(LAYER.WG).insert(path)
```

I was concerned about there being potential gaps when joining a straight and a curved path, but there were no visible gaps when viewing in KLayout.
